controllers/common.go

```diff
@@ -25,3 +61,55 @@ func ReconcileBatch(l logr.Logger, reconcileFuncs ...ReconcileFunc) (bool, error
 	}
 	return true, nil
 }
+
+func PopulateResticSecret(name string, namespace string, label string) (*corev1.Secret, error) {
+
+
+	// define Restic secret to be created
+	newResticSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-secret", name),
+			Namespace: namespace,
+			Labels: map[string]string{
+				label: name,
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	return newResticSecret, nil
+}
+
+func BuildResticSecret(givensecret *corev1.Secret, secret *corev1.Secret, resticrepo string) error {
+
+	if resticrepo == "" {
```

Comment:
Instead of adding this error here, I think we should add a `validateResticSecret`, WDYT ?

Reply:
this repo is different than the one is restic-secret. its the new repo we create. I am inclined towards leaving this here instead of creating a func just for this.

Reply:
I thought of bunching this together with the secret [params](https://github.com/konveyor/volume-snapshot-mover/pull/102#discussion_r905407573), not proposing a function for just one param.

Reply:
I am not sure on how to group it together. Can you explain?